modules/core/src/test/java/org/apache/synapse/config/xml/TaskManagerFactoryTest.java

```diff
@@ -0,0 +1,90 @@
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+package org.apache.synapse.config.xml;
+
+import org.apache.axiom.om.OMElement;
+import org.apache.axiom.om.util.AXIOMUtil;
+import org.apache.synapse.SynapseException;
+import org.apache.synapse.startup.quartz.QuartzTaskManager;
+import org.apache.synapse.task.TaskManager;
+import org.junit.Assert;
+import org.junit.Test;
+
+import javax.xml.stream.XMLStreamException;
+
+/**
+ * This is the test class for TaskManagerFactory class.
+ */
+public class TaskManagerFactoryTest {
+
+    /**
+     * test createTaskManager
+     *
+     * @throws XMLStreamException - XMLStreamException.
+     */
+    @Test
+    public void testCreateTaskManager() throws XMLStreamException {
+        String inputXML = "<taskManager provider=\"org.apache.synapse.startup.quartz.QuartzTaskManager\"/>";
+        OMElement element = AXIOMUtil.stringToOM(inputXML);
+        TaskManager taskManager = TaskManagerFactory.createTaskManager(element, null);
+        Assert.assertTrue("TaskManager is not created.", taskManager instanceof QuartzTaskManager);
+    }
+
+    /**
+     * test createTaskManager with parameter specified for OMElement
+     *
+     * @throws XMLStreamException - XMLStreamException.
+     */
+    @Test
+    public void testCreateTaskManager2() throws XMLStreamException {
+        String inputXML = "<taskManager provider=\"org.apache.synapse.startup.quartz.QuartzTaskManager\">"
+                + "<parameter xmlns=\"http://ws.apache.org/ns/synapse\"  name =\"test\"> test </parameter>"
+                + "</taskManager>";
+        OMElement element = AXIOMUtil.stringToOM(inputXML);
+        TaskManager taskManager = TaskManagerFactory.createTaskManager(element, null);
+        Assert.assertTrue("TaskManager is not created.", taskManager instanceof QuartzTaskManager);
+    }
+
+    /**
+     * test createTaskManager with wrong providerName
+     *
+     * @throws XMLStreamException - XMLStreamException.
+     */
+    @Test(expected = SynapseException.class)
+    public void testCreateTaskManager3() throws XMLStreamException {
+        String inputXML = "<taskManager provider=\"org.apache.synapse.startup.quartz.QuartzTaskManagerDummy\">"
+                + "<parameter xmlns=\"http://ws.apache.org/ns/synapse\"  name =\"test\"> test </parameter>"
+                + "</taskManager>";
+        OMElement element = AXIOMUtil.stringToOM(inputXML);
+        TaskManagerFactory.createTaskManager(element, null);
+    }
+
+    /**
+     * test createTaskManager with provider not specified.
+     *
+     * @throws XMLStreamException - XMLStreamException.
+     */
+    @Test(expected = SynapseException.class)
```

Comment:
Can we check for a specific assert message as well? If not how are we going to differentiate from testCreateTaskManager3() and  testCreateTaskManager4()  as they both throw the same Exception Type?

Reply:
Fixed this. Thank you